packages/redux/src/users/addresses/actions/factories/removeUserDefaultBillingAddressFactory.ts

```diff
@@ -0,0 +1,65 @@
+import * as actionTypes from '../../actionTypes.js';
+import {
+  type Config,
+  type DeleteUserDefaultBillingAddress,
+  toBlackoutError,
+  type User,
+} from '@farfetch/blackout-client';
+import { getDefaultAddress } from '../../reducer.js';
+import type { Dispatch } from 'redux';
+import type { RemoveUserDefaultBillingAddressAction } from '../../types/index.js';
+import type { StoreState } from '../../../../index.js';
+
+/**
+ * Responsible for deleting the users default billing address.
+ *
+ * @param deleteUserDefaultBillingAddress - Delete default Billing address client.
+ *
+ * @returns Thunk factory.
+ */
+const removeUserDefaultBillingAddressFactory =
+  (deleteUserDefaultBillingAddress: DeleteUserDefaultBillingAddress) =>
+  (userId: User['id'], config?: Config) =>
+  async (
+    dispatch: Dispatch<RemoveUserDefaultBillingAddressAction>,
+    getState: () => StoreState,
+  ): Promise<number> => {
+    const addressId = getDefaultAddress(
```

Comment:
I think it is better to have all the code inside the try/catch block.

Reply:
The addressId this need to be outside of the try catch, so it can be used in the try and catch clauses. This file has been updated, please validate.

The addressId now is fetched outside of the try but the error thrown is thrown inside of the try clause.

Reply:
You can declare the variable with a let and set it to undefined and have the assignment inside the try / catch block. In that way, you will have that code covered as well by the try/catch.

Reply:
Agreed, changed it 👍